static/js/pm_list.js

```diff
@@ -1,52 +1,99 @@
 import $ from "jquery";
+import _ from "lodash";
+
+import render_more_private_conversations from "../templates/more_pms.hbs";
 
-import * as narrow_state from "./narrow_state";
-import * as people from "./people";
 import * as pm_list_data from "./pm_list_data";
 import * as pm_list_dom from "./pm_list_dom";
+import * as resize from "./resize";
 import * as stream_popover from "./stream_popover";
 import * as ui from "./ui";
 import * as ui_util from "./ui_util";
 import * as vdom from "./vdom";
 
 let prior_dom;
-let private_messages_open = false;
 
 // This module manages the "Private messages" section in the upper
 // left corner of the app.  This was split out from stream_list.js.
 
+// To keep a track of the state of private messages section.
+let private_messages_collapsed = false;
+
+// To keep a track whether we're zoomed In or not.
+let zoomed = false;
+
+export function show_more_private_conversations() {
+    zoomed = true;
+}
+
+export function hide_more_private_conversations() {
+    zoomed = false;
+}
+
 function get_filter_li() {
-    return $(".top_left_private_messages .private_messages_header");
+    return $("#private_messages_section #private_messages_list .private-messages-section-header");
 }
 
-function set_count(count) {
+export function set_count(count) {
     ui_util.update_unread_count_in_dom(get_filter_li(), count);
 }
 
 function remove_expanded_private_messages() {
     stream_popover.hide_topic_popover();
-    ui.get_content_element($("#private-container")).empty();
+    // While making the private messages list empty
+    // we check for any active conversation thread present in it and
+    // if yes, then append it below the collapsed PM section.
+    const $active_conversation_thread = $(".expanded_private_messages li.active-sub-filter");
+    $(".expanded_private_messages").empty().append($active_conversation_thread);
 }
 
 export function close() {
```

Comment:
I think this doesn't have to be exported anymore

Reply:
Yeah! Updated the PR with this change too.